### PR TITLE
Add spec metadata fields section to K8S secret store docs

### PR DESCRIPTION
**Pull request by nyemade-uversky**
Added Spec metadata fields section and removed the "name" metadata attribute from code example

Signed-off-by: nyemade-uversky <55847877+nyemade-uversky@users.noreply.github.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [X] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description
Added Spec metadata fields section and removed the "name" metadata attribute from code example

## Issue reference
#2389

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md [hunk 1]
```diff
@@ -30,8 +30,11 @@ spec:
   type: secretstores.kubernetes
   version: v1
   metadata:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md]: This needs to be 
  metadata: []

see this as an example https://v1-8.docs.dapr.io/reference/components-reference/supported-state-stores/setup-inmemory/#component-format

File: daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md [hunk 2]
```diff
@@ -30,8 +30,11 @@ spec:
...
-  - name: ""
 ```
+
+## Spec metadata fields
+For the Kubernetes secret store component there are no additional metadata attributes that need to be set during setup.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md]:
```suggestion
For the Kubernetes secret store component, there are no metadata attributes.
```

Comment (hhunter-ms) [reply daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md]: For me, just add a comma after "component".

Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md]: updated

**Review by hhunter-ms — APPROVED** [review]
lgtm

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**